src/python/views/demo_tigergraph_circle/__init__.py

```diff
@@ -0,0 +1,774 @@
+import asyncio
+import graphistry
+import os
+import pandas as pd
+import streamlit as st
+from components import GraphistrySt, URLParam
+from neptune_helper import gremlin_helper, df_helper
+from css import all_css
+from util import getChild
+import time
+import altair as alt
+from TigerGraph_helper import tg_helper
+import plotly.express as px
+
+from gremlin_python import statics
+from gremlin_python.process.graph_traversal import __
+from gremlin_python.process.traversal import WithOptions, T
+
+############################################
+#
+#   DASHBOARD SETTINGS
+#
+############################################
+#  Controls how entrypoint.py picks it up
+
+
+app_id = 'tigergraph_circle'
+logger = getChild(app_id)
+urlParams = URLParam(app_id)
+node_id_col = 'id'
+src_id_col = 'from_id'
+dst_id_col = 'to_id'
+node_label_col = 'Source_Type'
+edge_label_col = 'Destination_Type'
+
+# Setup a structure to hold metrics
+metrics = {'tigergraph_time': 0, 'graphistry_time': 0,
+           'node_cnt': 0, 'edge_cnt': 0, 'prop_cnt': 0}
+
+conn = tg_helper.connect_to_tigergraph()
+
+
+# Define the name of the view
+def info():
+    return {
+        'id': app_id,
+        'name': 'TigerGraph: Fraud Filter circle',
+        'tags': ['demo', 'tigergraph_demo_circle']
+    }
+
+
+def run():
+    run_all()
+
+
+############################################
+#
+#   PIPELINE PIECES
+#
+############################################
+
+
+# Have fun!
+def custom_css():
+    all_css()
+    st.markdown(
+        """<style>
+
+        </style>""", unsafe_allow_html=True)
+
+
+# Given URL params, render left sidebar form and return combined filter settings
+# https://docs.streamlit.io/en/stable/api.html#display-interactive-widgets
+def sidebar_area():
+    # q = conn.runInstalledQuery("mostDirectInfections")
+    # Most_infectious_IDS = q[0]['Answer']
+    # MI_List = [d['v_id'] for d in Most_infectious_IDS if 'v_id' in d]
+    num_edges_init = urlParams.get_field('num_matches', 0.5)
+    # MI_List.reverse()
+    idList = [i for i in range(1, 500)]
+    st.sidebar.header("TigerGraph Anti-Fraud")
+    tg_host = st.sidebar.text_input ('TigerGraph Host')
+    tg_username = st.sidebar.text_input ('TigerGraph Username')
+    tg_password = st.sidebar.text_input ('TigerGraph Password')
+    tg_graphname = st.sidebar.text_input ('TigerGraph Graphname')
+    tg_secret = st.sidebar.text_input('TigerGraph Secret')
+    
+    if st.sidebar.button("Connect"):
+        try:
+            conn = tg.TigerGraphConnection(host=tg_host, graphname=tg_graphname, username=tg_username, password=tg_password)
```

Comment:
are there any lifecycle/leak implications of conn? e..g, should it be done as a `with` statement during the run-filter?

Reply:
In the `run_all` function, it is first checked whether `sidebar_area` returns a value. If it does, which means conn is valid, then the `run_filters` is called. Otherwise the function returns.